pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1372,7 +1372,11 @@ func (l *LibvirtDomainManager) GetGuestInfo() (v1.VirtualMachineInstanceGuestAge
 // InterfacesStatus returns the interfaces Guest Agent reported
 func (l *LibvirtDomainManager) InterfacesStatus(domainInterfaces []api.Interface) []api.InterfaceStatus {
 	interfaces := l.agentData.GetInterfaceStatus()
-	return agentpoller.MergeAgentStatusesWithDomainData(domainInterfaces, interfaces)
+	if interfaces != nil {
+		return agentpoller.MergeAgentStatusesWithDomainData(domainInterfaces, interfaces)
```

Comment:
I would add unit test here as well to make sure that in case `l.agentData.GetInterfaceStatus()` is need the `Merge...` is not invoked.

Reply:
It means that we need to redesign, is it ok ?
because i can just mock InterfacesStatus but not its internals ?

Reply:
No need to redesign, you just need to check the return value of `InterfacesStatus`. It should be nil.